File: Classes/UIImage+EXIFCompensation.m
//
//  UIImage+EXIFCompensation.m
//  xkcd
//
//  Created by Paul Rehkugler on 5/31/13.
//
//

#import "UIImage+EXIFCompensation.h"

@implementation UIImage (EXIFCompensation)

- (CGSize) exifAgnosticSize {
	size_t bitmapWidth = CGImageGetWidth(self.CGImage);
	size_t bitmapHeight = CGImageGetHeight(self.CGImage);
	
	return CGSizeMake(bitmapWidth, bitmapHeight);
}

@end

File: Classes/UIImage+EXIFCompensation.swift
//
//  UIImage+EXIFCompensation.swift
//  xkcd
//
//  Created by Paul Rehkugler on 1/24/16.
//
//

import UIKit

extension UIImage {

    /// The size of an image, without accounting for its EXIF orientation.
    @objc var exifAgnosticSize: CGSize {
		if let cgImage = cgImage {
			return CGSize(width: CGFloat(cgImage.width), height: CGFloat(cgImage.height));
		}
		
		return size;
    }
}
